### DISCUSSION TITLE
Scrcpy Popout window ratio question

### DISCUSSION THREAD

**Discussion by CorruptionVR** [yume-chan/ya-webadb#546] [Q&A]
hey, @yume-chan, I have a question about the scrcpy video popout. are you going to make it so it can only be one size, or are you going to make it so that when you resize the window, it'll keep its ratio?

**Comment by yume-chan** [comment]
It will be same as Scrcpy, screen resolution won't change, letterbox (black bars) will be used when aspect ratio doesn't match.

Unless Scrcpy supports such a feature, I can't add it myself, so please send your feature request to Scrcpy. But AFAIK, Android's support for changing screen resolution isn't great, there is no event for that, so screen record API won't change output video resolution when that happens.

WSA (Windows Subsystem for Android), on the other hand, uses a similar approach to RemoteApps. The Android emulator has the same screen size as the physical display, Apps are displayed as windows, and cropped into individual windows on Windows desktop. However that should also be impossible on real Android devices because desktop mode support varies between vendors.

**Reply by CorruptionVR** [reply]
no, I meant the browser window's size, not the actual device's resolution. If you have used Android Studio's AVD, when you resize one edge, the entire window resizes to keep showing the whole screen, while keeping the aspect ratio the same.

**Reply by yume-chan** [reply]
It's not possible to keep aspect ratio while resizing browser windows, so as the first paragraph said, it will be same as Scrcpy, window can be resized freely, and black bars will be added.

**Reply by CorruptionVR** [reply]
@yume-chan, are you absolutely sure it isn't possible?
